numpy/_core/src/multiarray/number.c

```diff
@@ -363,7 +366,12 @@ fast_scalar_power(PyObject *o1, PyObject *o2, int inplace, PyObject **result)
     }
 
     PyArrayObject *a1 = (PyArrayObject *)o1;
-    if (!(PyArray_ISFLOAT(a1) || PyArray_ISCOMPLEX(a1))) {
+    if (PyArray_ISOBJECT(a1)) {
```

Comment:
Thanks, yeah this seems right/safe to me.  I am not sure I see why `**2` confuses a custom `__array_ufunc__`, but I don't think this is an issue (and probably restores old behavior).  (If we had a bigger reason to keep it as is, I would be tempted to ask downstream to fix it there.)

Curious, does `**0.5` always end up with `sqrt()`?

---

(Maybe the nicer solution for at least `**2` would actually be to make the object `square` implementation use `**2` internally then this special case wouldn't even be needed.  But no priority so if, let's do that as a follow-up.)